In [4]:
import smtplib
from email.message import EmailMessage

In [5]:
smtp_gmail = smtplib.SMTP('smtp.gmail.com', 587)

In [6]:
#서버 연결 설정
smtp_gmail.ehlo()

#서버 연결 암호화
smtp_gmail.starttls()

#로그인
smtp_gmail.login('이메일@gmail.com','AppPassword') #AppPassword가 없다면 myaccount.google.com/apppassword로 가서 2-step 인증을 통해 부여받아야한다. 

(235, b'2.7.0 Accepted')

로그인이 성공했다면 '235, b'2.7.0 Accepted' 라는 메세지가 뜬다.

로그인시 에러가 발생한다면 보안 허용이 낮은 수준으로 되지 않은 경우이다.  'smtplib.SMTPAuthenticationError: (534~'와 같은 오류 메세지가 뜬다.

https://myaccount.google.com/lesssecureapps 의 <보안수준이 낮은 앱>에서 "보안 수준이 낮은 앱 허용"이 "사용"으로 바꾸면 된다. 


In [8]:
msg = EmailMessage()

msg['Subject'] = "안녕" #제목 입력

msg.set_content("이거 파이썬으로 보낸거야 (smtplib)") #내용 입력

msg['From'] = '보내는이@gmail.com'#보내는 사람

msg['To'] = '받는이@gmail.com' #받는 사람

smtp_gmail.send_message(msg)

{}